blocks/editable/format-toolbar.js

```diff
@@ -151,17 +152,24 @@ class FormatToolbar extends Component {
 						className="editable-format-toolbar__link-modal"
 						style={ linkStyle }
 						onSubmit={ this.submitLink }>
+						<label
```

Comment:
@afercia 
I think it makes sense to use aria-label in this specific environment because it simplifies the implementation by quite a bit. 
My only hope is that JAWS has support for aria, just like how the other popular screen readers do.